src/components/ProtectedRoute/ProtectedRoute.tsx

```diff
@@ -0,0 +1,23 @@
+import { useIdentify } from 'hooks/useIdentify';
+import { Navigate, Outlet, useLocation } from 'react-router-dom';
+import { AppRoutes } from 'types/AppRoutes';
+
+export function ProtectedRoute() {
+  const [userData, { isLoading }] = useIdentify();
+  const { pathname } = useLocation();
+
+  if (isLoading) {
+    return null;
+  }
+
+  return userData ? (
+    <Outlet />
+  ) : (
+    <Navigate
+      to={AppRoutes.signIn}
+      state={{
+        from: pathname,
+      }}
+    />
+  );
```

Comment:
❗️
В таких кейсах ранний `return` всегда лучше, чем тернарка
```suggestion
  if (userData) {
    return <Outlet />
  }
  
  return <Navigate
        to={AppRoutes.signIn}
        state={{
          from: pathname,
        }}
      />
```